### ISSUE TITLE
Mysql support and .net core

### ISSUE THREAD

**Issue by Jorerim** [signumsoftware/framework#273]
Hi,
this framework looks very promising and I am giving it a try. 

1. Can you please explain why this framework does not supports MySQL / PostgreSQL  ? I have experience with  SQL server, in 2 out of 3 apps we have have migrated MySQL. the cluster gets disconnected all the time and we end up spending a lot of money on DBAs. MySQL was more stable and we paid 0 for DBAs and 0 for licenses.  I mean the license is so expensive... So, do you have a plan to go in the MySQL / PostgreSQL ?
Bottom line, paying thousands of dollars every month for database that has better free alternatives... its a bit pain in the ass.

2. Do you supports row level security? is it documented?

3. Do you supports workflow for automating activities in the database? is it documented?

4. I saw a post here about the .net core support. I don't see in the commits that it was implemented. Do you still have plans to do so?

Thanks.

**Comment by olmobrutall** [comment]
Hi @Jorerim,

I'm very glad that you want to try it. If you find some difficulties ask here. The documentation is not very good/up to date, specially for the React part. But once you get how think works I don't think there is a more productive framework for building line of business apps. 

Answering your questions: 

1. There is no support because I haven't develop it. But there are no fundamental reasons that make it impossible. Is also not so trivial because, on the human side, right now I'm the only one that is able to work on the LINQ provider, and the company I'm working on uses SQL Server, so even if I implement a MySQL/PostgreSQL at home, if I don't use it at work it will always be a second-class citizen. There are also some technical issues: 

    * CROSS/OUTER APPLY: the LINQ provider requires this extensions of SQL to translate something like: 
    ```CSharp
     from c in Database.Query<Customer>()
      from inv in c.Invoices().OrderBy(a=>a.CreationDate).Take(5)
      select new {c, inv}
      ```
      AFAIK MySql has no equivalent, but Postgree does have one I think. 
     *  SqlDependency for cache invalidation. I think Postgre has an equivalent but not sure about MySql. 
     *  SqlHierarchyId for hierachical structures. Only in Folder Extension.  
     * TemoralTables to keep historical data. Only in TimeMachine Extension (Quite new).
    So, all in all, this could be done when I work in a project where the saving in licencing justifies the investment (about 2-4 weeks of work?), or someone tries to make the changes (I could help) of is willing to pay me to do it. 

2. Yes, it's called TypeConditions. (https://github.com/signumsoftware/southwind/blob/master/Southwind.Logic/Starter.cs#L181) You register the `where` condition with a particular name (TypeConditionSymbol) for 1 or more entities (tables). Then you can use this conditions to change the Authorization rules for each specific role. The only documentation available is an old Word from 2012. Some thinks have changed but is mostly ok. 
[Part6Authorization.docx](https://github.com/signumsoftware/framework/files/1967270/Part6Authorization.docx)


3. There is a Workflow extension following BPMN standard, more for human activities than database scripts or so. For doing stuff in the database there are also Scheduled Tasks (repetitive tasks) , Processes (long-running tasks), C# Migrations (after a deployment) and simple methods in a Console application.

4. I've been working in .Net Core support lately. It compiles but doesn't run yet, I need some stuff in .Net Core 2.1 (SqlDependency support, ActionResult<T>) but I faced some issues probably due to being in preview 2. I assume that next week .Net Core 2.1 final will be presented in Build conference, and there will be a working version at the end of this month. For some time there will be two branches, master and masterCore but moving forward there will be only .Net Core.  

Tell me if you face any other problem / doubt. 

Regards.

**Comment by Jorerim** [comment]
Thank you Olmo.
The MySQL is disappointing but I get your point, its kind of one man show for now. 
Out of curiosity, why does your company continue working with SQL server? short and long run it is not cost effective.

I will generate an environment in your website to test it out.
Can you please tell me something I need to know before I run it the first time?
I understand that I need to run the generate environment script and create userConnections file

Something about React? Visual Studio?
Is it smooth process to start from scratch?
 
Thanks

**Comment by olmobrutall** [comment]
In theory could be done in 30 mins or so but there are always issues because there are many parts involved: sql server, visual Studio, IIS, asp.net, git, Typescript, React, webpack plus signum specific stuff. 

But it’s worth, at the end you haven a full stack app working that does a lot of things already. 

 Follow the instructions here: http://www.signumsoftware.com/en/DuplicateApplication, 

Don’t worry about the specific versions, just take the latest one, and tell me where you get stuck

And tell me what

**Comment by olmobrutall** [comment]
Hi again @jorerim, i just pushen an Updates Version of Southwind, with the latest framework, Extensions, typescript and react. 

Probably is a good idea to create a new application again now. 

Regards.

**Comment by olmobrutall** [comment]
Hi @Jorerim, .Net Core is ready: https://github.com/signumsoftware/framework/commit/2997541863847ad35cc2a51e97e5c1f2c58f8050#comments
